protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -120,8 +120,9 @@ def _new_project_config(
 
     @staticmethod
     def _validate_project_name(name: str):
-        pattern = r"^[a-zA-Z0-9][a-zA-Z0-9_]*$"
+        # https://github.com/software-mansion/scarb/blob/main/scarb/src/core/package/name.rs#LL42C9
+        pattern = r"^(?!_$)[a-zA-Z_][0-9a-zA-Z_]*$"
         if not re.match(pattern, name):
             raise ProtostarException(
-                f"Provided project name {name} does not match regex {pattern}. Choose a different project name."
+                f"Provided project name {name} does not match the regex {pattern}. Choose a different project name."
```

Comment:
I think the pattern used here is complicated enough that the error message with it is not useful in any way. I'd either updated it to use a different message, or consider a different way of name validation.